python/grass/temporal/abstract_space_time_dataset.py

```diff
@@ -1517,45 +1513,30 @@ def get_registered_maps_as_objects(
             "contains": maps that contain (fully cover) the provided spatial extent
 
         :return: The ordered map object list,
-                In case nothing found None is returned
+                In case nothing found an empty list is returned
         """
 
         dbif, connection_state_changed = init_dbif(dbif)
 
         obj_list = []
 
-        # Older temporal databases have no bottom and top columns
-        # in their views so we need a work around to set the full
-        # spatial extent as well
-
-        rows = get_tgis_metadata(dbif)
-        db_version = 0
-
-        if rows:
-            for row in rows:
-                if row["key"] == "tgis_db_version":
-                    db_version = int(float(row["value"]))
-
         # use all columns
         rows = self.get_registered_maps(
             None, where, order, dbif, spatial_extent, spatial_relation
         )
 
-        if rows is not None:
-            has_bt_columns = False
-            has_semantic_label = False
-            first_row = True
-            for row in rows:
-                if first_row:
-                    first_row = False
-                    # check keys in first row
-                    # note that 'if "bottom" in row' does not work
-                    # because row is not a dict but some db backend object
-                    if "bottom" in row.keys() and "top" in row.keys():
-                        has_bt_columns = True
-                    if "semantic_label" in row.keys():
-                        has_semantic_label = True
+        if rows:
+            # Older temporal databases have no bottom and top columns
+            # in their views so we need a work around to set the full
+            # spatial extent as well
+
+            # check keys in first row
+            # note that 'if "bottom" in row' does not work
+            # because row is not a dict but some db backend object
+            has_bt_columns = "bottom" in rows[0].keys() and "top" in rows[0].keys()
```

Comment:
When I read specifically about the sqlite3 Row class (in Python), a row can be accessed both by index or column name. So technically `rows[0]["bottom"]` could work, and if so something like the following might work (I'm not sure if it's the right test to do with it to have what you want, but an illustration of what it could look like without two other iterations for membership testing, probably only impactful if there are a lot of columns)

```suggestion
            has_bt_columns = rows[0]["bottom"] is not None  and rows[0]["top"] is not None
```

Reply:
The purpose of this check is to get the data structure, i.e. if the columns "bottom" and "top" are present in the metadata DB view. If "bottom" is present "top" is present too, so checking for "bottom" should be enough...

Reply:
Does the code break if there isn't the two columns?

Reply:
Yes, I think so. I do have some more fixes and will look into `compute_datetime_delta` too, ASAP.

Reply:
After refreshing up on the data structures in Python (most of that knowledge dated from the late-Python 2 to Python 3 transition era, so it was a bit far), my initial first impressions on the cost of a membership testing are more relaxed now. If we are talking about dictionaries, `.keys()` returns a dictionary view and thus I suppose calling `in` is like a hash-based lookup, or something that would be similar to what is done for sets. So there wouldn't be a problem even for the original.

https://docs.python.org/3/library/stdtypes.html#dict.keys

https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects